cli/kata-check_amd64.go

```diff
@@ -15,44 +19,107 @@ const (
 	archCPUModelField  = genericCPUModelField
 )
 
+// CPU types
+const (
+	CPUtypeIntel   = 0
+	CPUtypeAMD     = 1
+	CPUtypeUnknown = -1
+)
+
+// CPUtype save the CPU type
+var CPUtype int
+
 // archRequiredCPUFlags maps a CPU flag value to search for and a
 // human-readable description of that value.
-var archRequiredCPUFlags = map[string]string{
-	"vmx":    "Virtualization support",
-	"lm":     "64Bit CPU",
-	"sse4_1": "SSE4.1",
-}
+var archRequiredCPUFlags map[string]string
 
 // archRequiredCPUAttribs maps a CPU (non-CPU flag) attribute value to search for
 // and a human-readable description of that value.
-var archRequiredCPUAttribs = map[string]string{
-	"GenuineIntel": "Intel Architecture CPU",
-}
+var archRequiredCPUAttribs map[string]string
 
 // archRequiredKernelModules maps a required module name to a human-readable
 // description of the modules functionality and an optional list of
 // required module parameters.
-var archRequiredKernelModules = map[string]kernelModule{
-	"kvm": {
-		desc: "Kernel-based Virtual Machine",
-	},
-	"kvm_intel": {
-		desc: "Intel KVM",
-		parameters: map[string]string{
-			"nested": "Y",
-			// "VMX Unrestricted mode support". This is used
-			// as a heuristic to determine if the system is
-			// "new enough" to run a Kata Container
-			// (atleast a Westmere).
-			"unrestricted_guest": "Y",
-		},
-	},
-	"vhost": {
-		desc: "Host kernel accelerator for virtio",
-	},
-	"vhost_net": {
-		desc: "Host kernel accelerator for virtio network",
-	},
+var archRequiredKernelModules map[string]kernelModule
+
+func init() {
```

Comment:
What's the reason for running this in the very special `init()` function ?

Reply:
Did not know there were any restrictions on using function init(). Checked go language documentation looking for function init() restrictions, but could not find any. Function init() seemed like a good place to set up these variables. 

If need be, the call to this function can be made from main().

Reply:
As an alternative you could move this into a new function and call it from the anonymous `Action` function defined in `kataCheckCLICommand`.